### ISSUE TITLE
Loss starts from initial value every epoch for structured classifier

### ISSUE THREAD

**Issue by SirJohnFranklin** [keras-team/autokeras#928]
### Bug Description
Loss starts from initial value every epoch for structured classifier:

> Train for 16 steps, validate for 4 steps
> 1/16 [>.............................] - ETA: 1s - loss: 2.3992 - accuracy: 0.8438
>  4/16 [======>.......................] - ETA: 0s - loss: 2.6361 - accuracy: 0.8281
>  7/16 [============>.................] - ETA: 0s - loss: 2.6707 - accuracy: 0.8259
> 10/16 [=================>............] - ETA: 0s - loss: 2.4452 - accuracy: 0.8406
> 13/16 [=======================>......] - ETA: 0s - loss: 2.5466 - accuracy: 0.8341
> 16/16 [==============================] - 1s 36ms/step - loss: 2.5186 - accuracy: 0.8347 - val_loss: 2.2202 - val_accuracy: 0.8560
> Epoch 2/1000
>  1/16 [>.............................] - ETA: 0s - loss: 2.3992 - accuracy: 0.8438
>  4/16 [======>.......................] - ETA: 0s - loss: 2.6361 - accuracy: 0.8281
>  7/16 [============>.................] - ETA: 0s - loss: 2.6707 - accuracy: 0.8259
> 10/16 [=================>............] - ETA: 0s - loss: 2.4452 - accuracy: 0.8406
> 13/16 [=======================>......] - ETA: 0s - loss: 2.5466 - accuracy: 0.8341
> 16/16 [==============================] - 0s 28ms/step - loss: 2.5186 - accuracy: 0.8347 - val_loss: 2.2202 - val_accuracy: 0.8560
> Epoch 3/1000
>  1/16 [>.............................] - ETA: 0s - loss: 2.3992 - accuracy: 0.8438
>  4/16 [======>.......................] - ETA: 0s - loss: 2.6361 - accuracy: 0.8281
>  7/16 [============>.................] - ETA: 0s - loss: 2.6707 - accuracy: 0.8259
> 10/16 [=================>............] - ETA: 0s - loss: 2.4452 - accuracy: 0.8406
> 13/16 [=======================>......] - ETA: 0s - loss: 2.5466 - accuracy: 0.8341
> 16/16 [==============================] - 0s 28ms/step - loss: 2.5186 - accuracy: 0.8347 - val_loss: 2.2202 - val_accuracy: 0.8560
> Epoch 4/1000
>  1/16 [>.............................] - ETA: 0s - loss: 2.3992 - accuracy: 0.8438
>  4/16 [======>.......................] - ETA: 0s - loss: 2.6361 - accuracy: 0.8281
>  7/16 [============>.................] - ETA: 0s - loss: 2.6707 - accuracy: 0.8259
> 10/16 [=================>............] - ETA: 0s - loss: 2.4452 - accuracy: 0.8406
> 13/16 [=======================>......] - ETA: 0s - loss: 2.5466 - accuracy: 0.8341
> 16/16 [==============================] - 0s 27ms/step - loss: 2.5186 - accuracy: 0.8347 - val_loss: 2.2202 - val_accuracy: 0.8560
> Epoch 5/1000

### Reproducing Steps
Run titanic example and set verbose to 1

### Expected Behavior
Decreasing loss for every epoch, approaching some value

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 3.7
 - autokeras: 1.0.0
 - tensorflow: '2.1.0'

### Additional context
Code to reproduce:

```
import autokeras as ak
import tensorflow as tf

import pandas as pd

if __name__ == '__main__':
    print(f"Tensorflow version: {tf.__version__}")

    x_train = pd.read_csv('titanic/train.csv')
    print(type(x_train))  # pandas.DataFrame
    y_train = x_train.pop('survived')
    print(type(y_train))  # pandas.Series

    # Preparing testing data.
    x_test = pd.read_csv('titanic/eval.csv')
    y_test = x_test.pop('survived')

    # Initialize the structured data classifier.
    clf = ak.StructuredDataClassifier(max_trials=20)  # It tries 20 different models.
    # Feed the tensorflow Dataset to the classifier.
    clf.fit(x_train, y_train, use_multiprocessing=False, batch_size=32, validation_split=0.2)
    # Predict with the best model.
    predicted_y = clf.predict(x_test)
    # Evaluate the best model with testing data.
    print(clf.evaluate(x_test, y_test))
```

**Comment by kmjthe** [comment]
I have the same problem.,,,,
